cocotb_test/simulator.py

```diff
@@ -336,14 +433,22 @@ def get_parameter_commands(self, parameters):
         parameters_cmd = []
         for name, value in parameters.items():
             parameters_cmd.append("-P")
-            parameters_cmd.append(self.toplevel + "." + name + "=" + str(value))
+            # TODO: How should parameters be handled when using multiple top-level designs?
+            parameters_cmd.append(self.toplevel[0] + "." + name + "=" + str(value))
 
         return parameters_cmd
 
     def compile_command(self):
+        if self.verilog_compile_args is not None:
+            self.compile_args += self.verilog_compile_args
+
+        toplevel = []
+        for t in self.toplevel:
+            toplevel += ["-s", t]
 
         cmd_compile = (
-            ["iverilog", "-o", self.sim_file, "-D", "COCOTB_SIM=1", "-s", self.toplevel, "-g2012"]
+            ["iverilog", "-o", self.sim_file, "-D", "COCOTB_SIM=1", "-g2012"]
```

Comment:
```suggestion
            ["iverilog", "-o", self.sim_file, "-D", "COCOTB_SIM=1", "-g2012"]
```
I know it was there before, but... can we unconditionally add `-g2012` here? Will it not collide when a user wants to add another `-g` option?

Reply:
I never used additional flags. Looking at `iverilog -h` makes it seem that it can accept multiple `-g` arguments though. This can be achieved by using `compile_args`.